pootle/static/js/shared/utils/time.test.js

```diff
@@ -0,0 +1,507 @@
+/*
+ * Copyright (C) Pootle contributors.
+ *
+ * This file is a part of the Pootle project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import expect from 'expect';
+import lolex from 'lolex';
+import { describe, it } from 'mocha';
+
+import { relativeTime, relativeTimeMessage, timeDelta } from './time';
+
+
+describe('timeDelta', () => {
+  it('returns an empty structure for invalid datetimes', () => {
+    const undefinedDelta = {
+      isFuture: null,
+      seconds: null,
+      minutes: null,
+      hours: null,
+      days: null,
+      weeks: null,
+      months: null,
+      years: null,
+    };
+    expect(timeDelta('')).toEqual(undefinedDelta);
+    expect(timeDelta(null)).toEqual(undefinedDelta);
+    expect(timeDelta(undefined)).toEqual(undefinedDelta);
+    expect(timeDelta('Invalid')).toEqual(undefinedDelta);
+    expect(timeDelta('2016-14-12')).toEqual(undefinedDelta);
+    expect(timeDelta('2016-12-12 25:02:11')).toEqual(undefinedDelta);
+  });
+
+  describe('positive duration', () => {
+    const tests = [
+      {
+        unit: 'seconds',
+        now: '2016-07-07T00:00:00+00:00',
+        args: ['2016-07-07T00:00:44+00:00'],
+        expected: {
+          isFuture: true,
+          seconds: 44,
+          minutes: 0,
+          hours: 0,
+          days: 0,
+          weeks: 0,
+          months: 0,
+          years: 0,
+        },
+      },
+      {
+        unit: 'seconds (TZ-aware)',
+        now: '2016-07-07T00:00:00+02:00',
+        args: ['2016-07-06T22:00:44+00:00'],
+        expected: {
+          isFuture: true,
+          seconds: 44,
+          minutes: 0,
+          hours: 0,
+          days: 0,
+          weeks: 0,
+          months: 0,
+          years: 0,
+        },
+      },
+      {
+        unit: 'minutes',
+        now: '2016-07-07T00:00:00+00:00',
+        args: ['2016-07-07T00:59:00+00:00'],
+        expected: {
+          isFuture: true,
+          seconds: 3540,
+          minutes: 59,
+          hours: 0,
+          days: 0,
+          weeks: 0,
+          months: 0,
+          years: 0,
+        },
+      },
+      {
+        unit: 'minutes (TZ-aware)',
+        now: '2016-07-07T00:00:00+02:00',
+        args: ['2016-07-06T22:59:00+00:00'],
+        expected: {
+          isFuture: true,
+          seconds: 3540,
+          minutes: 59,
+          hours: 0,
+          days: 0,
+          weeks: 0,
+          months: 0,
+          years: 0,
+        },
+      },
+      {
+        unit: 'minutes',
```

Comment:
Does this, and the next test provide anything else that the two previous tests are not?